client/src/components/RIC-Widget/OutfitList.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState, useCallback } from 'react';
+import ProductCard from './ProductCard.jsx';
+
+function OutfitList({ productId }) {
+  const listState = 'outfit';
+  const [ifOutfit, setExists] = useState(false);
+  const [outfitItems, setOutfit] = useState([]);
+
+  function addItem(id) {
+    if (outfitItems.indexOf(id) === -1) {
+      setOutfit(prevArray => [...prevArray, id]);
+      setExists(true);
+    }
+  };
+
+  const triggerDelete = useCallback((index) => {
+    let currentOutfits = outfitItems.map((item) => item);
+    currentOutfits.splice(index, 1);
+
+    setOutfit(currentOutfits);
+
+    if (currentOutfits.length === 0) {
+      setExists(false);
+    };
+  });
+
+  return (
+    <div>
+      <div className='AddCard' onClick={() => addItem(productId)}>
+        <div>+</div>
+        <div>Add to Outfit</div>
+      </div>
+      <div>
+        {ifOutfit
+          ? <div> {outfitItems.map((id, index) => (
+            <ProductCard
+              productId={id} key={index}
+              index={index}
```

Comment:
I don't think it's necessary to have the index on here twice if you've already put it as the key. I'm pretty sure it's possible to utilize the key attribute with getAttribute('key')